pageserver/src/page_service.rs

```diff
@@ -340,20 +340,22 @@ impl PageServerHandler {
                         };
 
                         let zenith_fe_msg = PagestreamFeMessage::parse(copy_data_bytes)?;
+                        let tenant_id = tenantid.to_string();
+                        let timeline_id = timelineid.to_string();
```

Comment:
Same as in get_page_at_lsn, these could be pretty expensive.

Reply:
Thanks, I'll check if we can hold references for these.

Reply:
This is tricky as they're passed in page request. Still it's stingy() cost that's concerning here right?

Reply:
Benchtest results from release build on my dev setup. 
Used test_small_seqscans to get some numbers. I think it's covering in-mem pages and don't have any ancillary side  effects. 

Baseline (without metrics collection)
```
------------------------------------------------------------------- Benchmark results ----
test_small_seqscans[vanilla-100000].table_size: 3629056 bytes
test_small_seqscans[vanilla-100000].run: 21.418 s
test_small_seqscans[zenith-100000].table_size: 3629056 bytes
test_small_seqscans[zenith-100000].run: 48.036 s
``` 

With empty tags (no stringify)
```
------------------------------------------------------------------- Benchmark results ----
test_small_seqscans[vanilla-100000].table_size: 3629056 bytes
test_small_seqscans[vanilla-100000].run: 21.662 s
test_small_seqscans[zenith-100000].table_size: 3629056 bytes
test_small_seqscans[zenith-100000].run: 47.755 s
```

With tags
```
------------------------------------------------------------------- Benchmark results ----
test_small_seqscans[vanilla-100000].table_size: 3629056 bytes
test_small_seqscans[vanilla-100000].run: 21.659 s
test_small_seqscans[zenith-100000].table_size: 3629056 bytes
test_small_seqscans[zenith-100000].run: 48.720 s
```

Reply:
Added custom encoding for hex strings, 
```
mainline:
zid.to_string           time:   [237.75 ns 239.61 ns 241.77 ns]                 
patched:
zid.to_string           time:   [87.561 ns 90.342 ns 93.693 ns]
```